src/explainer.ts

```diff
@@ -6,11 +6,11 @@ import * as pluralize from 'pluralize';
 import * as shelljs from 'shelljs';
 
 import { formatComplex, formatOne, Typed, formatType } from "./schema-formatting";
-import { getActiveKubeconfig, getUseWsl } from './components/config/config';
+import { getKubeconfig, getUseWsl } from './components/config/config';
 
 async function loadKubeconfig(): Promise<kubernetes.KubeConfig> {
     const kc = new kubernetes.KubeConfig();
-    const kubeconfig = getActiveKubeconfig();
+    const kubeconfig = getKubeconfig();
     if (kubeconfig) {
```

Comment:
>if (kubeconfig) { [](start = 4, length = 17)

Similar to my other comment: following the change, this is always going to be true. Given that we have specific cases to handle the absence of a value in vscode.workspace.getConfiguration(EXTENSION_CONFIG_KEY)[KUBECONFIG_PATH_KEY], I assume this is a behavior we intend to keep.
@itowlson > Could you please provide guidance/context on this? Should we keep using vscode.workspace.getConfiguration(EXTENSION_CONFIG_KEY)[KUBECONFIG_PATH_KEY] here?

Reply:
The check here appears to be because of needing to get the kubeconfig from a Windows user's Linux home directory if they have specified to use the WSL version of the tools.  The logic is:

* If the user has overridden kubeconfig in extension settings, then that's what they should get - if they have also asked for WSL then they should jolly well be specifying a WSL file path.
* If the user _hasn't_ overriden kubeconfig in extension settings, and is on WSL, we _don't_ want to use the Windows default path, because that path won't work in WSL (and won't be pointing at the same kubeconfig that other kubectl commands, which _are_ WSLised, use).  We want to use the same default that kubectl uses on Linux (Linux $KUBECONFIG or home dir).
* Otherwise, we use the default that kubectl uses on Windows (Windows %KUBECONFIG% or home dir)

So yes, we need to keep this behaviour.  It's kind of ugly having to drop down to the config-level API from this high level but it may be the most pragmatic design - maybe add a TODO for it and move on.

However, this does make me think there is a potentially nasty can of worms at the API level: consumer extensions which use the `KubectlV1.*` functions get consistent behaviour guaranteed, but consumers who use the kubeconfig path _need to know whether that path relates to the host OS or to WSL_.  It may be that this isn't a consideration for your extension - I don't know what your use case for the path is.  But I think it needs to be visible so that extension authors at least need to handle the possibility, if only to say "sorry, user, we don't offer this feature if you're using the WSL versions of the tools."

So maybe the API needs to return something like an `{ isHostPath: true; path: string } | { isHostPath: false; guestPathType: 'wsl'; guestPath: string }` so that consumers can't naively use the path without at least considering whether and how to handle the WSL scenario...?